manager/volume.go

```diff
@@ -280,16 +277,18 @@ func (m *VolumeManager) Attach(name, nodeID string, disableFrontend bool) (v *lo
 		return nil, fmt.Errorf("volume %v is restoring data", name)
 	}
 
-	// already desired to be attached
-	if v.Spec.NodeID != "" {
-		if v.Spec.NodeID != nodeID {
-			return nil, fmt.Errorf("Node to be attached %v is different from previous spec %v", nodeID, v.Spec.NodeID)
-		}
+	// no need to error if we are already attached to this node
+	if v.Status.State == types.VolumeStateAttached && v.Spec.NodeID == nodeID {
```

Comment:
Need to make sure all the parameters are the same as requested, e.g. `DisableFrontend`